modules/Auth/Auth.cpp

```diff
@@ -18,7 +18,14 @@ void Auth::init() {
 
     for (const auto& token_provider : this->r_token_provider) {
         token_provider->subscribe_provided_token([this](ProvidedIdToken provided_token) {
-            std::thread t([this, provided_token]() { this->auth_handler->on_token(provided_token); });
+            std::thread t([this, provided_token]() {
+                this->p_main->publish_token_validation_status(types::authorization::TokenValidationStatus::Processing);
+                const auto res = this->auth_handler->on_token(provided_token);
+                if (res == TokenHandlingResult::REJECTED) {
+                    this->p_main->publish_token_validation_status(types::authorization::TokenValidationStatus::Rejected);
+                }
+                this->p_main->publish_token_validation_status(types::authorization::TokenValidationStatus::Idle);
+                });
             t.detach();
```

Comment:
Maybe I'm missing something here, but isn't the reason, that this subscription handler spawns a new thread, because there can come in multiple tokens at a time?  If so, how should one distinguish these different tokens and the processing/rejected/idle events?

Reply:
@wku12 unfortunately I don't know about how the interface should look like, so I can't comment on this.  I was merely focusing on logic

Reply:
@a-w50 
What I meant here is that I acknowledged your concern and tried to rationalize my misunderstanding of the code and the mutex purpose.

I did the changes described in the previous comment, so the message published right now also contains the ProvidedIdToken object, so the TokenValidationStatus is now linked to the token. Therefore my last sentence was kindly asking you to review it again :)